docs/require-by-string-aliases.md

```diff
@@ -131,7 +131,7 @@ If `.luaurc` contained the following `paths` array:
 
 Then, `module.luau` could simply require `dependency.luau` like this:
 ```lua
-local dependency = require("dependency")
+local dependency = require("@dependency")
```

Comment:
I don't think this is right, `@dependency` looks through the `aliases` map, and should never undergo resolution through the `paths`, right? So this is an error.

Reply:
I agree, and not sure why github has hidden my feedback :( https://github.com/luau-lang/rfcs/commit/0643bda137b1dc990cf782dd6e7f99d173c253ff#r134719538

Reply:
Done.